special/Mk/diffs/Uses_dlang.mk.diff

```diff
@@ -0,0 +1,381 @@
++#		Minimal required d-runtime version (inclusive) (default:20791)
++# - DRT_MAXVER
++#		Maximum allowed d-runtime version (inclusive) (default:20999)
++# - DLANG_DEBUG
++#		Print extra debug output for USES=dlang
++#
++# for ldc:
++# - D_REQ_LLVM 	LLVM version to require (60 70 80)
++#
++# for gdc:
++# - D_REQ_GCC	GCC version to require (8 9 10)
++#
++# Variables provided by dlang.mk
++# for any:
++# - CHOSEN_DCOMPILER_TYPE
++#	D compiler selected
++# - DCOMPILER
++#	Full path to D-compiler selected.
++# - DCOMPILER_DMD
++#	Full path to DMD interface compatible compiler selected.
++# - DRUNTIME_VERSION
++#	D runtime version of selected D-compiler.
++# - DFLAGS
++#	Addional DFLAGS variables to be passed to the D-compiler.
++# for ldc:
++# - LLVM_CONFIG
++#	Full path to llvm-config being used
++# - LLVM_VER
++# 	Major version of llvm being used
++# for gdc:
++# - GCC_VER
++# 	Major version of gcc being used
++#
++# MAINTAINER: ddegroot@talon.nl
++
++.if !defined(_INCLUDE_USES_DLANG_MK) || ${_INCLUDE_USES_DLANG_MK} == recheck
++
++# locally defined variables
++CHOSEN_DCOMPILER_TYPE=
++D_INCLUDE_DIR?=	${LOCALBASE}/include/d
++D_INCLUDE_DIRS?=${D_INCLUDE_DIR}
++D_LIB_DIR?=	${LOCALBASE}/lib
++D_LIB_DIRS?=	${D_LIB_DIR}
++
++# tmp variables
++_DFLAGS_LDC?=
++_DFLAGS_GDC?=
++_DFLAGS_DMD?=-I${D_INCLUDE_DIR} -L${D_LIB_DIR}
++
++_USES_POST+=	dlang
++VALID_ARGS=	ldc gdc dmd bootstrap
++
++# Binary Lookup
++ldc_BINARY!=	${WHICH} ldc2 || true
++gdc_BINARY!=	${WHICH} gdc || true
++dmd_BINARY!=	${WHICH} dmd || true
++
++# Defaults
++DRT_MINVER?=	20791
++DRT_MAXVER?=	20999
++
++#
++# Check USE_XXX
++#
++.if defined(USE_LDC)
++DCOMPILERS=ldc
++.endif
++.if defined(USE_GDC)
++DCOMPILERS=gdc
++.endif
++.if defined(USE_DMD)
++DCOMPILERS=dmd
++.endif
++
++DCOMPILERS?=ldc gdc dmd
++
++#
++# Check compiler availability
++#
++.for i in ${DCOMPILERS}
++.	if empty(CHOSEN_DCOMPILER_TYPE)
++.		if !empty(${i}_BINARY)
++_D_VERSION!=${${i}_BINARY} --version | ${SED} -En 's|.*DMD.* v([0-9]).([0-9]+).([0-9]).*|\1\2\3|p'
++.			if !empty(_D_VERSION) && (${_D_VERSION} >= ${DRT_MINVER}) && (${_D_VERSION} <= ${DRT_MAXVER})
++CHOSEN_DCOMPILER_TYPE=${i}
++DRUNTIME_VERSION=${_D_VERSION}
++.			endif
++.		endif
++.	endif
++.endfor
++
++#
++# bootstrap requested or required
++# if none were found, use the first one in the DCOMPILERS list as the
++# fallback
++# if bootstrapping, we have recheck d-compiler availability on next run (so
++# undo the include guard at the top)
++#
++.if empty(CHOSEN_DCOMPILER_TYPE)
++.	if ${dlang_ARGS} == bootstrap
++.		if ${PORTNAME} == ldc
++BUILD_DEPENDS+=	${LOCALBASE}/ldc-ltsmaster/bin/ldmd2:lang/ldc-bootstrap
++DCOMPILER=		${LOCALBASE}/ldc-ltsmaster/bin/ldc2
++DCOMPILER_DMD=		${LOCALBASE}/ldc-ltsmaster/bin/ldmd2
++_INCLUDE_USES_DLANG_MK=	recheck
++.		elif ${PORTNAME} == gdc
++USE_GCC?=		8+
++OPTIONS_DEFAULT+=	FORCE_BOOTSTRAP
++_INCLUDE_USES_DLANG_MK=	recheck
++.		else
++BUILD_DEPENDS+= ${LOCALBASE}/bin/ldc2:lang/ldc${LLVM_FLAVOR:D@${LLVM_FLAVOR}}
```

Comment:
Please remove the flavors as we agreed

Reply:
Yes i will have to redesign quite a bit of it.

Reply:
Would the other dports be excepted if these changes were made, if so i can try to make some time and finish this up. If not and freebsd ports would have to be excepted first, then it might be better to drop this and move on.